File: src/lib/services/examination.ts
import { invoke } from "@tauri-apps/api/core";

export interface Examination {
    id?: number;
    patient_id: number;
    exam_date: string;
    complaint?: string;
    findings?: string;
    diagnosis?: string;
    treatment?: string;
}

export interface CreateExaminationInput {
    patient_id: number;
    complaint?: string;
    findings?: string;
    diagnosis?: string;
    treatment?: string;
}

export interface Prescription {
    id?: number;
    exam_id: number;
    medication_name: string;
    dosage?: string;
    frequency?: string;
}

export interface CreatePrescriptionInput {
    exam_id: number;
    medication_name: string;
    dosage?: string;
    frequency?: string;
}

export const examinationService = {
    async create(input: CreateExaminationInput): Promise<number> {
        return await invoke("create_examination", { input });
    },

    async getByPatient(patientId: number): Promise<Examination[]> {
        return await invoke("get_patient_examinations", { patientId });
    },

    async createPrescription(input: CreatePrescriptionInput): Promise<number> {
        return await invoke("create_prescription", { input });
    },

    async getPrescriptions(examId: number): Promise<Prescription[]> {
        return await invoke("get_examination_prescriptions", { examId });
    }
};

File: src-tauri/src/models/examination.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Examination {
    pub id: Option<i64>,
    pub patient_id: i64,
    pub exam_date: String,
    pub complaint: Option<String>,
    pub findings: Option<String>,
    pub diagnosis: Option<String>,
    pub treatment: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct CreateExaminationInput {
    pub patient_id: i64,
    pub complaint: Option<String>,
    pub findings: Option<String>,
    pub diagnosis: Option<String>,
    pub treatment: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Prescription {
    pub id: Option<i64>,
    pub exam_id: i64,
    pub medication_name: String,
    pub dosage: Option<String>,
    pub frequency: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct CreatePrescriptionInput {
    pub exam_id: i64,
    pub medication_name: String,
    pub dosage: Option<String>,
    pub frequency: Option<String>,
}
